### PR TITLE
TRACING-4062 | Add OpenTelemetry journald receiver

### PR DESCRIPTION
**Pull request by pavolloffay**
Version(s):

4.12, 4.13, 4.14, 4.15, 4.16

This PR should be published as part of RHOSDT 3.2 release

Issue:

https://issues.redhat.com/browse/TRACING-4062

Link to docs preview:

https://74207--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-configuration-of-otel-collector.html#journald-receiver_otel-configuration-of-otel-collector

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/otel-collector-components.adoc [hunk 1]
```diff
@@ -199,6 +199,117 @@ The OpenCensus receiver provides backwards compatibility with the OpenCensus pro
...
+  tolerations:
+  - key: node-role.kubernetes.io/master
+    operator: Exists
+    effect: NoSchedule
+EOF
+----
+<1> Filter output by message priorities or priority ranges. Defaults to `info`.
+<2> A list of units to read entries from. Empty list will read all units.
+<3> A very long logs and logs with unprintable characters will also be included. Defaults to `false`.
+<4> The receiver will pause reading a file and attempt to resend the current batch of logs if it encounters an error from downstream components. Defaults to `fals`.
```
Comment (IshwarKanse) [inline modules/otel-collector-components.adoc:308]: Small typo. fals > false.

File: modules/otel-collector-components.adoc [hunk 2]
```diff
@@ -199,6 +199,117 @@ The OpenCensus receiver provides backwards compatibility with the OpenCensus pro
...
+  name: system:openshift:scc:privileged
+subjects:
+- kind: ServiceAccount
+  name: privileged-sa
+  namespace: otel-journald
+---
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: otel-joural-logs
```
Comment (IshwarKanse) [inline modules/otel-collector-components.adoc:243]: joural > journal

**Comment by max-cx** [comment]
@pavolloffay, feel free to close this draft because I opened https://github.com/openshift/openshift-docs/pull/75257.

**Pull request closed without merging**